ludwig/trainers/trainer_llm.py

```diff
@@ -434,8 +435,28 @@ def evaluation(self, dataset, dataset_name, metrics_log, batch_size, progress_tr
         predictor = LlmFineTunePredictor(
             self.model, batch_size=batch_size, distributed=self.distributed, report_tqdm_to_ray=self.report_tqdm_to_ray
         )
-        metrics, _ = predictor.batch_evaluation(dataset, collect_predictions=False, dataset_name=dataset_name)
+        metrics, _, inp_tar_out_dict = predictor.batch_evaluation(
+            dataset, collect_predictions=False, dataset_name=dataset_name
+        )
+
+        tokenizer = self.dist_model.tokenizer
+
+        llm_eval_examples = {"inputs": [], "targets": [], "outputs": []}
+        for key in inp_tar_out_dict["inputs"]:
+            for inp in inp_tar_out_dict["inputs"][key]:
+                inp = remove_left_padding(inp, tokenizer)[0]
+                llm_eval_examples["inputs"].append(tokenizer.decode(inp))
+
+        for key in inp_tar_out_dict["targets"]:
+            for tar in inp_tar_out_dict["targets"][key]:
+                tar = remove_left_padding(tar, tokenizer)[0]
+                llm_eval_examples["targets"].append(tokenizer.decode(tar))
```

Comment:
Could we do `tokenizer.decode(tar, skip_special_symbols=True)` to avoid manually removing left padding?